pandora-client-web/src/components/characterRestrictionOverride/characterRestrictionOverride.tsx

```diff
@@ -0,0 +1,213 @@
+				restrictionOverride != null ? (
+					<CharacterRestrictionOverrideLeave
+						{ ...restrictionOverride }
+						doModeExit={ doModeExit }
+						processing={ processing }
+					/>
+				) : (
+					<>
+						<p>
+							<i>You are currently not in safemode or timeout mode.</i>
+						</p>
+						<p>
+							<strong>Warning:</strong> After entering safemode, you will not be able to leave it for { FormatTimeInterval(GetRestrictionOverrideConfig('safemode').allowLeaveAt) }!<br />
+							(Timeout mode can toggled on off without any time restrictions.)
+						</p>
+						<Row padding='medium' alignX='space-between'>
+							<Button onClick={ hide }>Cancel</Button>
+							<Button onClick={ doTimeoutModeEnter } disabled={ processing }>
+								Enter timeout mode!
+							</Button>
+							<Button onClick={ doSafeModeEnter } disabled={ processing }>
+								Enter safemode!
+							</Button>
+						</Row>
+					</>
+				)
+			}
+		</ModalDialog>
+	);
+}
+
+export function GetRestrictionOverrideText(type: RestrictionOverride['type']): string {
+	switch (type) {
+		case 'safemode':
+			return 'safemode';
+		case 'timeout':
+			return 'timeout mode';
+	}
+}
+
+function CharacterRestrictionOverrideLeave({ type, allowLeaveAt, doModeExit, processing }: RestrictionOverride & {
+	doModeExit: () => void;
+	processing: boolean;
+}): ReactElement {
+	const currentTime = useCurrentTime();
+	const canLeave = currentTime >= allowLeaveAt;
+	const mode = GetRestrictionOverrideText(type);
+	const { hide } = useRestrictionOverrideDialogContext();
+
+	return (
+		<>
+			<p>
+				<strong>You are currently in a { mode }!</strong><br />
+				{
+					canLeave ? null : <>You need to wait { FormatTimeInterval(allowLeaveAt - currentTime) } before you can leave the { mode }.</>
+				}
+			</p>
+			<Row padding='medium' alignX='space-between'>
+				<Button onClick={ hide }>Cancel</Button>
+				<Button
+					disabled={ !canLeave || processing }
+					className='fadeDisabled'
+					onClick={ doModeExit }
+				>
+					Leave { mode }
+				</Button>
+			</Row>
+		</>
+	);
+}
+
+export function CharacterRestrictionOverrideWarningContent({ mode }: { mode?: RestrictionOverride; }): ReactElement | null {
+	if (!mode)
+		return null;
+
+	let HelpText: () => ReactElement;
+
+	switch (mode.type) {
+		case 'safemode':
+			HelpText = CharacterSafemodeHelpText;
+			break;
+		case 'timeout':
+			HelpText = CharacterTimeoutModeHelpText;
+			break;
+	}
+
+	return (
+		<span className='safemode'>
+			This character is in { GetRestrictionOverrideText(mode.type) }
+			<ContextHelpButton>
+				<HelpText />
+				<p>
+					As general hint: You can find safemode and timeout mode for your character in the top left menu, by clicking on<br />
+					your character's name.
+				</p>
+			</ContextHelpButton>
+		</span>
+	);
+}
+
+function CharacterSafemodeHelpText(): ReactElement {
+	return (
+		<>
+			<p>
+				Safemode is a mode in which items do not apply any restrictions to the character in it.<br />
```

Comment:
I propose to make the first sentence here now bold to be consistent with the later sentence explaining timeout mode.